### ISSUE TITLE
Use nvm.el

### ISSUE THREAD

**Issue by scottaj** [scottaj/mocha-mode#1]

**Comment by scottaj** [comment]
This is pretty low priority. The current technique of just specifying which node to use works really well.

**Comment by dcluna** [comment]
In case someone needs a quick fix for this, I have something like the following in my `.dir-locals.el`:

```elisp
((js2-mode . ((eval . (progn
                       (setq mocha-which-node
                             (if (boundp 'nvm-current-version)
                                 (concat (cadr nvm-current-version) "/bin/node")
                               (concat (getenv "HOME") "/.nvm/versions/node/v" (s-chomp (f-read-text (concat default-directory ".nvmrc"))) "/bin/node" )))
                       ))
;; other vars here...
              )))
```

Notice that I'm relying on this file being in the root of the project, the same place where `.nvmrc` is located.

**Comment by Fuco1** [comment]
@dcluna I've replaced `default-directory` with `(locate-dominating-file default-directory "package.json")` so it works from nested directories as well.  I like the solution, at least for a while.

**Comment by felipeochoa** [comment]
If you're using [`nvm.el`](https://github.com/rejeep/nvm.el), once [this PR](rejeep/nvm.el#12) is merged, you can add something like this to your init file:

```elisp
(add-hook 'js2-mode-hook 'nvm-use-for-buffer)
```

Then, leaving `mocha-which-node` as `node` will work if you've defined an `.nvmrc` file, without having to mess around with dir locals (particularly eval)
